nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -45,22 +70,57 @@ class ServiceClient
     typename RequestType::SharedPtr & request,
     const std::chrono::seconds timeout = std::chrono::seconds::max())
   {
-    auto result_future = client_->async_send_request(request);
+    while (!client_->wait_for_service(std::chrono::seconds(1))) {
+      if (!rclcpp::ok()) {
+        throw std::runtime_error(
+                "ServiceClient: service call interrupted while waiting for service");
+      }
+      RCLCPP_DEBUG(node_->get_logger(), "Waiting for service to appear...");
```

Comment:
Can we change the RCLCPP_DEBUG to RCLCPP_INFO and add the name of the service, so the user will know if we're hanging and why

Reply:
Sure, changed.